lib/puppet/module/task.rb

```diff
@@ -61,6 +62,50 @@ def self.is_tasks_filename?(path)
       return true
     end
 
+    def self.get_file_details(path)
+      unless File.absolute_path(path) == File.path(path)
+        msg = _("File pathnames cannot include relative paths")
+        raise InvalidMetadata.new(msg, 'puppet.tasks/invalid-metadata')
+      end
+
+      name = path.split("/").last
```

Comment:
Without seeing the changes in puppetserver, I'm not sure if this is ok. I think we'll want the original path as specified in metadata so we can uniquely match an entry in metadata to the entry in the list of files.

Reply:
I guess for files from a directory, what that should be is unclear. I'd expect it to be the original prefix, plus the part added by the glob.